docs/en/04_Changelogs/4.0.0.md

```diff
@@ -28,12 +85,187 @@ Note that this will not allow you to utilise certain file versioning features in
 
 	:::yaml
 	\SilverStripe\Filesystem\Flysystem\FlysystemAssetStore:
-		legacy_paths: true
+	  legacy_paths: true
 
 
 See [/developer_guides/files/file_management] for more information on how the new system works.
 
 
+### Upgrade code which acts on `Image`
+
+As all image-specific manipulations has been refactored from `Image` into an `ImageManipulations` trait, which 
+is applied to both `File` and `DBFile`. These both implement a common interface `AssetContainer`, which
+has the `getIsImage()` method. In some cases, it may be preferable to invoke this method to detect
+if the asset is an image or not, rather than checking the subclass, as the asset may also be a `DBFile` with
+an image filter applied, rather than an instance of the `Image` dataobject.
+
+In addition, a new file category `image/supported` has been added, which is a subset of the `image` category.
+This is the subset of all image types which may be assigned to the `[api:Image]` dataobject, and may have
+manipulations applied to it. This should be used as the file type restriction on any `[api:UploadField]` which
+is intended to upload images for manipulation. All image categories have also been specified via constants
+on the `[api:File]` DataObject.
```

Comment:
A category constant could be created on a class other than `File`, but then you still need to duplicate its string value in `File.app_categories` YAML - unless you use `Config::inst()->update()` where you can refer to constants. The constant use just feels like a half-hearted attempt at making a config schema, providing a false sense of security and making it more likely that config keys get out of sync in projects (hard to "global find and replace" due to different notations). We can throw exceptions whenever an invalid key is used, just not through static analysis.